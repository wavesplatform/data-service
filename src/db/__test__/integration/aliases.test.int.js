const { Nothing } = require('folktale/maybe');

const db = require('./createDb')();

const ADDRESS = '3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy';

describe('Aliases', () => {
  it('should return Maybe(alias) for one correctly', done => {
    db.aliases
      .one('sexy-boys')
      .run()
      .listen({
        onResolved: maybeX => {
          expect(maybeX).toMatchSnapshot();
          done();
        },
      });

    db.aliases
      .one('NON_EXISTING_ALIAS')
      .run()
      .listen({
        onResolved: maybeX => {
          expect(maybeX).toEqual(Nothing());
          done();
        },
      });
  });

  describe('request by address', () => {
    it('should return correct data if requested without showBroken', done => {
      db.aliases
        .many({ address: ADDRESS })
        .run()
        .listen({
          onResolved: mxs => {
            expect(mxs).toMatchSnapshot();
            done();
          },
        });
    });

    it('should return correct data if requested with showBroken', done => {
      db.aliases
        .many({
          address: ADDRESS,
          showBroken: true,
        })
        .run()
        .listen({
          onResolved: mxs => {
            expect(mxs).toMatchSnapshot();
            done();
          },
        });
    });
  });
});
