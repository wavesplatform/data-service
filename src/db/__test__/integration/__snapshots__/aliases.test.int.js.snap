// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aliases request by address should return correct data if requested with showBroken 1`] = `
Array [
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "carl",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "lmao",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "ubiq",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "payments",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "wings",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "matt",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "jonathan",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "platinum",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "silver",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "cobalt",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "hydrogen",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "zinc",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "swift",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "rnsd",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "happy",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "flux",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "abram",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "fortuna",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "gamb",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "first",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "fortune",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "national",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "haha",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "whenmoon",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "music",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "gaming",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "rubles",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "dollars",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "rail",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "surf",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "homo",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "noah",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "emma",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "emily",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "peace",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "vitamin",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "whalepool",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "eternal",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "tsunami",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "rocket",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "solid",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": null,
        "alias": "scam",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "hack",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "when",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "nudes",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "email",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "mail",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "messi",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "star",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "allah",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "jesus",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "banker",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "flight",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "bird",
      },
    },
  },
]
`;

exports[`Aliases request by address should return correct data if requested without showBroken 1`] = `
Array [
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "carl",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "lmao",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "ubiq",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "payments",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "wings",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "matt",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "jonathan",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "platinum",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "silver",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "cobalt",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "hydrogen",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "zinc",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "swift",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "rnsd",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "happy",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "flux",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "abram",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "fortuna",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "gamb",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "first",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "fortune",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "national",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "haha",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "whenmoon",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "music",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "gaming",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "rubles",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "dollars",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "rail",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "surf",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "homo",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "noah",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "emma",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "emily",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "peace",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "vitamin",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "whalepool",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "eternal",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "tsunami",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "rocket",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "solid",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "hack",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "when",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "nudes",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "email",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "mail",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "messi",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "star",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "allah",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "jesus",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "banker",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "flight",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": Object {
        "address": "3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy",
        "alias": "bird",
      },
    },
  },
]
`;

exports[`Aliases should return Maybe(alias) for one correctly 1`] = `
Object {
  "@@tag": "Just",
  "@@type": "folktale:Maybe",
  "@@value": Object {
    "value": Object {
      "address": "3PQqCXNPoTRsHRK2nZyXWENaZwLmxrUZPBi",
      "alias": "sexy-boys",
    },
  },
}
`;
